sandpack-react/src/contexts/sandpackContext.tsx

```diff
@@ -406,12 +436,37 @@ class SandpackProvider extends React.PureComponent<
             delete this.unsubscribeQueuedListeners[clientId][listenerId];
           }
         };
+
+        return unsubscribeListener;
       }
     } else {
-      const unsubs = Object.values(this.clients).map((client) =>
+      // Push to the **global** queue
+      const listenerId = generateRandomId();
+      this.queuedListeners.global[listenerId] = listener;
+
+      // Add to the current clients
+      const clients = Object.values(this.clients);
+      const currentClientListeners = clients.map((client) =>
         client.listen(listener)
       );
-      return () => unsubs.forEach((unsub) => unsub());
+      const currentClientListenersUnsubscribe = () => {
+        currentClientListeners.forEach((unsubscribe) => unsubscribe());
+      };
+
+      const unsubscribeListener = () => {
+        const unsubscribeQueuedClients = Object.values(
+          this.unsubscribeQueuedListeners
+        );
+
+        unsubscribeQueuedClients.forEach((listenerOfClient) => {
```

Comment:
Can you leave a comment for this block as well? I believe it is unsubscribing the listeners which were initially queued globally and were added to each unsubscribeQueuedListeners[clientId], right?